.github/workflows/changelog.yml

```diff
@@ -1,25 +1,25 @@
-name: Check Unreleased
+name: Definition of Done
 on:
   pull_request:
-    types: [opened, edited]
-    branches-ignore:
-      - main
+    types: [opened, synchronize]
 permissions: {}
 
 jobs:
-  changelog:
+  check-changelog:
     runs-on: ubuntu-latest
     # Excluding Dependabot PRs from this check.
     if: github.actor != 'dependabot[bot]'
     steps:
-      - name: Checkout code
-        uses: actions/checkout@v2
+      - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
         with:
-          fetch-depth: 2
-
+          fetch-depth: 0
       - name: Check for new file in .unreleased
+        env:
+            BASE_BRANCH: ${{github.event.pull_request.base.ref}}
         run: |
-          FILE_COUNT=$(git diff --name-only main..HEAD -- .unreleased | wc -l)
+          git fetch origin $BASE_BRANCH:$BASE_BRANCH
```

Comment:
I think with `fetch-depth: 0` all branches should be already fetched, shouldn't they?

Reply:
Nope. It's still failing:

https://github.com/NordSecurity/libtelio/actions/runs/8829848747/job/24241755695?pr=511

Reply:
That's interesting